packages/core/core/src/factories/apiFactory/types.ts

```diff
@@ -0,0 +1,51 @@
+import { IntegrationPlugin } from './../../utils/nuxt';
+export interface ApiExtensionInstance {
+  beforeSetup?: (config, headers?: Record<string, string>) => any;
+  afterSetup?: ({ config, client }, headers?: Record<string, string>) => { config; client };
+  beforeCall?: ({ config, functionName, params }) => any;
+  afterCall?: ({ config, functionName, params }) => any;
+}
+
+export type ApiExtension = (req: any, res: any) => ApiExtensionInstance;
+
+export interface BaseApiFactoryParams<T, F = any> {
+  tag: string;
+  api: F;
+  isProxy?: boolean;
+}
+
+export interface ApiFactoryParams<T, F = any> extends BaseApiFactoryParams<T, F> {
+  onSetup: (config: T, headers?: Record<string, string>) => { config: T; client: any };
+  extensions?: ApiExtension[];
+}
+
+export interface ApiProxyFactoryParams<T, F = any> extends BaseApiFactoryParams<T, F> {
+  onSetup: (config: T, headers?: Record<string, string>) => { config: T };
+}
+
+export interface ApiInstance {
+  api: any;
+  client: any;
+  settings: any;
+}
+
+export interface FactoryInstance {
+  integrationPlugin: IntegrationPlugin;
+}
+
+export type CreateApiProxyFn = (givenConfig: any, customApi?: any) => ApiInstance
+export type CreateApiClientFn = (givenConfig: any, customApi?: any) => ApiInstance;
+
+export interface ProxyFactoryInstance extends FactoryInstance {
```

Comment:
be consistant, I see here all variations of words "factory" "API" and "proxy" used in different order and in different combinations

Let's stick to one naming for
- apiClientFactory
- apiProxyFactory
- a unified type for both

Reply:
unified type? they return different things and have different parameters

Reply:
then only 2 of those ;)